ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/internaldafny/extern/Com_Amazonaws_Kms.py

```diff
@@ -0,0 +1,58 @@
+import boto3
+import _dafny
+
+from botocore.config import Config
+from boto3.session import Session
+
+from standard_library.internaldafny.generated.Wrappers import Option_Some
+from com_amazonaws_kms.smithygenerated.com_amazonaws_kms.shim import KMSClientShim
+from com_amazonaws_kms.internaldafny.generated.Com_Amazonaws_Kms import *
+import com_amazonaws_kms.internaldafny.generated.Com_Amazonaws_Kms
+
+
+# Persist this across calls; this doesn't change
+available_aws_regions: list[str] = None
+
+def get_available_aws_regions():
+    global available_aws_regions
+    if available_aws_regions is not None:
+        return available_aws_regions
+    available_aws_regions = Session().get_available_regions("kms")
+    return available_aws_regions
+
+class default__(com_amazonaws_kms.internaldafny.generated.Com_Amazonaws_Kms.default__):
+    @staticmethod
+    def KMSClient(boto_client = None, region = None):
+        if boto_client is None:
+            if region is not None and region in get_available_aws_regions():
+                boto_config = Config(
+                    region_name=region
+                )
+                boto_client = boto3.client("kms", config=boto_config)
+            else:
+                boto_client = boto3.client("kms")
```

Comment:
nit: can we add a comment here saying that we will be dropped into the default region the client gives us?